File: LiSystem/LiAdmin/LiModel/Basic/TableModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using LiModel.Converter;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Serialization;
using LiModel.Form;

namespace LiModel.Basic
{
    public class TableModel : AGridlookUpEditModelBase
    {
        public override TEntity getInstance<TEntity>()
        {
            return new TEntity();
        }

        public static TableModel getInstanceByBasic()
        {
            TableModel tableModel = new TableModel();
            tableModel.dataBaseName = "LiSystem";
            tableModel.entityType = "Basic";
            tableModel.entityOrder = "master";
            tableModel.className = "JsonModel";
            tableModel.modifyDate = DateTime.Now;
            tableModel.datas = new List<ColumnModel>();

            return tableModel;
        }

        public override  string getValueMember()
        {
            return "entityKey";
        }

        public override string getDisplayMember()
        {
            return "entityName";
        }

        public override List<string> getSearchColumns()
        {

            if (!SearchColumns.Contains("entityName"))
            {
                SearchColumns.Add("entityName");
            }
            return SearchColumns;
        }

        /// <summary>
        /// 获取显示列
        /// </summary>
        public override List<string> getDisplayColumns()
        {
            if (!DisplayColumns.Contains("entityKey"))
            {
                DisplayColumns.Add("entityKey");
            }
            if (!DisplayColumns.Contains("entityName"))
            {
                DisplayColumns.Add("entityName");
            }
            return DisplayColumns;
        }

        /// <summary>
        /// 获取列名映射
        /// </summary>
        public override Dictionary<string, string> getDictModelDesc()
        {
            if (!DictModelDesc.ContainsKey("entityKey"))
            {
                DictModelDesc.Add("entityKey", "编码");
            }
            if (!DictModelDesc.ContainsKey("entityName"))
            {
                DictModelDesc.Add("entityName", "名称");
            }
            return DictModelDesc;
        }

        /// <summary>
        /// 
        /// </summary>
        public int id { set; get; }

        /// <summary>
        /// 所属数据库
        /// </summary>
        public string dataBaseName { set; get; }

        /// <summary>
        /// 实体类型，单据？档案？
        /// </summary>
        public string entityType { set; get; }

        /// <summary>
        /// 系统代码
        /// </summary>
        public string systemCode { set; get; }

        /// <summary>
        /// 比如单据Key
        /// </summary>
        public string entityKey { set; get; }

        /// <summary>
        /// 比如单据名称
        /// </summary>
        public string entityName { set; get; }

        /// <summary>
        /// 是否是主表master
        /// </summary>
        public string entityOrder { set; get; }

        /// <summary>
        /// 对应主键上的字段名
        /// </summary>
        public string entityColumnName { set; get; }

        /// <summary>
        /// 表名称
        /// </summary>
        public string tableName { set; get; }

        /// <summary>
        /// 表别名
        /// </summary>
        public string tableAliasName { set; get; }

        /// <summary>
        /// 表中文名
        /// </summary>
        public string tableAbbName { set; get; }

        /// <summary>
        /// 表描述
        /// </summary>
        public string tableDesc { set; get; }

        /// <summary>
        /// 类名
        /// </summary>
        public string className { set; get; }

        /// <summary>
        /// 主键名
        /// </summary>
        public string keyName { set; get; }

        /// <summary>
        /// 子表实体名
        /// </summary>
        public string childTableEntityColumnName { set; get; }

        /// <summary>
        /// 修改日期
        /// </summary>
        [JsonConverter(typeof(LiDateTimeConverter))]
        public DateTime modifyDate { set; get; }

        public List<ColumnModel> datas { set; get; }
    }
}

File: LiServer/Li_DRM/src/main/java/com/li/drm/model/TableModel.java
package com.li.drm.model;

import com.li.drm.annotate.*;
import com.li.drm.enumli.DatabaseGenerated;
import com.li.drm.enumli.Relationship;
import lombok.Getter;
import lombok.Setter;

import java.sql.Timestamp;
import java.util.List;

/**
 * 表信息模型，数据库模型
 */
@Getter
@Setter

@Table("TableInfo")
public class TableModel {
    @Key
    @DatabaseGeneratedOption(DatabaseGenerated.Identity)
    private Integer id;

    /**
     * 所属数据库
     */
    private String dataBaseName;

    /**
     * 实体类型，单据？档案？
     */
    private String entityType;
    /**
     * 比如单据Key
     */
    private String entityKey;

    /**
     * 比如单据名称
     */
    private String entityName;
    /**
     * 是否是主表master
     */
    private String entityOrder;

    /**
     * 对应主键上的字段名
     */
    private String entityColumnName;

    /**
     * 表名称
     */
    private String tableName;

    /**
     * 表别名
     */
    private String tableAliasName;

    /**
     * 表中文名
     */
    private String tableAbbName;

    /**
     * 表描述
     */
    private String tableDesc;

    /**
     * 类名
     */
    private String className;

    /**
     * 主键名
     */
    private String keyName;

    /**
     * 子表实体名
     */
    private String childTableEntityColumnName;

    /**
     * 修改日期
     */
    private Timestamp modifyDate;

    @RelationshipOption(Relationship.Many)
    @WillCascadeOnDelete(true)
    @RelationshipEntity(ColumnModel.class)
    private List<ColumnModel> datas;
}
